cranelift/codegen/src/isa/x64/inst/external.rs

```diff
@@ -131,7 +131,7 @@ fn enc_gpr(gpr: &Gpr) -> u8 {
     if let Some(real) = gpr.to_reg().to_real_reg() {
         real.hw_enc()
     } else {
-        unreachable!()
+        0
```

Comment:
Outdated change now?  (although I understand why this was done I think https://github.com/bytecodealliance/wasmtime/pull/10538 is the "true" fix? or something like that?)